### PR TITLE
Sketch of doing http.status log lines

### PR DESCRIPTION
**Pull request by cyx**
Was weighing between this approach and just manually doing log.SCountOne for each of the three methods (pub / sub / mkstream)

### CODE REVIEW

File: server/middleware.go [hunk 1]
```diff
@@ -24,3 +24,11 @@ func enforceHTTPS(ƒ http.HandlerFunc) http.HandlerFunc {
 		ƒ(w, r)
 	}
 }
+
+func logRequest(fn http.HandlerFunc) http.HandlerFunc {
```
Comment (tt) [inline server/middleware.go]: Perhaps use `ƒ` for consistency?

Comment (cyx) [inline server/middleware.go]: Ok, mostly just wondering if we should just go fn instead (linux's support
for special characters leaves something to be desired...)

I can always copy & paste it though :-)

File: util/response_logger.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+func (l *responseLogger) Header() http.Header {
+	return l.w.Header()
+}
+
+func (l *responseLogger) Write(b []byte) (int, error) {
+	if l.status == 0 {
+		// The status will be StatusOK if WriteHeader has not been called yet
+		l.status = http.StatusOK
```
Comment (tt) [inline util/response_logger.go]: Why not initialize with this value to avoid this check?

File: server/middleware.go (hunk 1 above)
Comment (tt) [inline server/middleware.go]: I'm fine changing to something that's easier to type.

**Comment by tt** [comment]
Looks good to me except for a couple very minor details.

File: util/response_logger.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	l.w.WriteHeader(s)
+	l.status = s
+}
+
+func (l *responseLogger) WriteLog() {
+	maskedStatus := strconv.Itoa(l.status/100) + "xx"
+
+	// TODO: possibly use heroku/slog for writing the
+	// structured logs instead
+	log.Printf("count#http.status.%s=1", maskedStatus)
```
Comment (catsby) [inline util/response_logger.go]: can we include the request id in the header, if present?

File: server/middleware.go (hunk 1 above)
Comment (catsby) [inline server/middleware.go]: We see `ƒ` in Endosome in a place or two. I would prefer `fn` personally...

**Comment by cyx** [comment]
Closing in favor of #23

**Pull request merged**